include/fi_util.h

```diff
@@ -542,10 +542,23 @@ struct util_wait_fd {
 	struct util_wait	util_wait;
 	struct fd_signal	signal;
 	fi_epoll_t		epoll_fd;
+	struct dlist_entry	fd_list;
+};
+
+typedef int (*ofi_wait_fd_try_func)(const void *arg);
```

Comment:
I think the try call implies the argument will be modified in some way.  An easy way to check this is to remove all casts from the try callback and declare any local variable as const.  Then see if a cast ends up being necessary to remove the const.